applications/StructuralMechanicsApplication/structural_mechanics_application.cpp

```diff
@@ -52,6 +52,9 @@ KratosStructuralMechanicsApplication::KratosStructuralMechanicsApplication():
     mShellThinCorotationalElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ), true ),
     // Adding the membrane element
     mMembraneElement3D3N( 0, Element::GeometryType::Pointer( new Triangle3D3 <Node<3> >( Element::GeometryType::PointsArrayType( 3 ) ) ) ),
+	mMembraneElement3D4N( 0, Element::GeometryType::Pointer( new Quadrilateral3D4 <Node<3> >( Element::GeometryType::PointsArrayType( 4 ) ) ), false ),	
```

Comment:
Indeed,Automated Differentiation, you can see examples in the FluidDynamicApplication, in the folder symbolic_generation, I am using too for contact, but it is my branch right now